x-pack/plugins/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -66,6 +76,40 @@ const statsRoute = createDatasetQualityServerRoute({
   },
 });
 
+const malformedDocsRoute = createDatasetQualityServerRoute({
+  endpoint: `GET ${DATA_STREAMS_MALFORMED_STATS_URL}`,
+  params: t.type({
+    query: t.intersection([
+      rangeRt,
+      dataStreamTypesRt,
+      t.partial({
+        datasetQuery: t.string,
+      }),
+    ]),
+  }),
```

Comment:
**suggestion:** it might be worth to also factor out this type in the common folder and re-use it to encode the query params when the client sends the request.